src/web/components/SharingPermission/SharingPermissionsTable.tsx

```diff
@@ -3,41 +3,59 @@ import { CheckedState } from '@radix-ui/react-checkbox';
 import { useEffect, useMemo, useState } from 'react';
 
 import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
-import { AvailableParticipantDTO } from '../../../api/routers/participantsRouter';
+import { useAvailableSiteList } from '../../services/site';
+import { formatStringsWithSeparator } from '../../utils/textHelpers';
 import { Dialog } from '../Core/Dialog';
+import { Loading } from '../Core/Loading';
 import { MultiSelectDropdown } from '../Core/MultiSelectDropdown';
 import { SortableTableHeader } from '../Core/SortableTableHeader';
 import { TriStateCheckbox, TriStateCheckboxState } from '../Core/TriStateCheckbox';
+import { isAddedByManual, MANUALLY_ADDED, SharingParticipant } from './bulkAddPermissionsHelpers';
 import { ParticipantsTable } from './ParticipantsTable';
 
 import './SharingPermissionsTable.scss';
 
 type SharingPermissionsTableProps = {
-  sharingParticipants: AvailableParticipantDTO[];
+  sharedSiteIds: number[];
+  sharedTypes: string[];
   onDeleteSharingPermission: (siteIds: number[]) => Promise<void>;
   participantTypes: ParticipantTypeDTO[];
 };
 
-function NoParticipant() {
-  return (
-    <div className='no-participants-container'>
-      <img src='/group-icon.svg' alt='group-icon' />
-      <div className='no-participants-text'>
-        <h2>No Participants</h2>
-        <span>You don&apos;t have any sharing permissions yet.</span>
-      </div>
-    </div>
-  );
-}
-
 export function SharingPermissionsTable({
-  sharingParticipants,
+  sharedSiteIds,
+  sharedTypes,
   onDeleteSharingPermission,
   participantTypes,
 }: SharingPermissionsTableProps) {
+  const { sites: availableParticipants, isLoading } = useAvailableSiteList();
   const [filterText, setFilterText] = useState('');
   const [checkedParticipants, setCheckedParticipants] = useState<Set<number>>(new Set());
   const [openConfirmation, setOpenConfirmation] = useState(false);
+  const siteIds = useMemo(() => new Set(sharedSiteIds), [sharedSiteIds]);
+
+  const sharingParticipants: SharingParticipant[] = useMemo(() => {
```

Comment:
Looking at all of the `useMemo` hooks we have in this function, there's a lot of cascading complexity - different `useMemo` results depending on each other. It's a sign to me the whole state needs to be re-thought. This is adding more complexity on top of something that already looks over-optimised.